src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -384,9 +387,13 @@ private void Cancel()
                 // When cancellation propagates, SendRequestBodyAsync will set _requestCompletionState to Failed
                 requestBodyCancellationSource?.Cancel();
 
-                if (sendReset)
+                // SendReset must be called under a lock and after a cancellation of requestBodyCancellationSource.
```

Comment:
Does it actually have to happen after we cancel requestBodyCancellationSource? It seems like the cancellation is going to propagate asynchronously anyway.

Reply:
> It seems like the cancellation is going to propagate asynchronously anyway.

That was my main concern about reordering. I'm not sure if it always propagates asynchronously. Need take a closer look.

Reply:
It seems cancellation callbacks can be [invoked on the canceling thread](https://github.com/dotnet/runtime/blob/91599fff26619cb8bccb7b39c39590e7cd9a4379/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs#L711) when there is no a synchronization context set, so I believe we shouldn't reorder calls here.

Reply:
The cancellation callback will be called on the canceling thread. But that doesn't guarantee anything about when cancellation propagates.

If you don't think we should change the ordering here, that's fine with me; I agree it's safer not to. But let's change or remove the comment because it makes it sound like the ordering is strictly necessary, when I don't think we know that for sure.

Reply:
OK. I removed the comment.